driver-core/src/main/java/com/datastax/driver/core/policies/HostFilterPolicy.java

```diff
@@ -0,0 +1,226 @@
+     * the session on which this plan need to be built. This can be {@code null} if
+     * the corresponding session has no keyspace logged in.
+     * @param statement the query for which to build a plan.
+     */
+    @Override
+    public Iterator<Host> newQueryPlan(String loggedKeyspace, Statement statement) {
+        // Just delegate to the child policy, since we filter the hosts not white
+        // listed upfront, the child policy will never see a host that is not white
+        // listed and thus can't return one.
+        return childPolicy.newQueryPlan(loggedKeyspace, statement);
+    }
+
+    @Override
+    public void onUp(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onUp(host);
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public void onSuspected(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onSuspected(host);
+    }
+
+    @Override
+    public void onDown(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onDown(host);
+    }
+
+    @Override
+    public void onAdd(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onAdd(host);
+    }
+
+    @Override
+    public void onRemove(Host host) {
+        if (predicate.apply(host))
+            childPolicy.onRemove(host);
+    }
+
+    @Override
+    public void close() {
+        if (childPolicy instanceof CloseableLoadBalancingPolicy)
+            ((CloseableLoadBalancingPolicy)childPolicy).close();
+    }
+    
+    /**
+     * Create a new policy that wraps the provided child policy but only "allow/forbids" hosts
+     * from the provided list.
+     *
+     * @param childPolicy the wrapped policy.
+     * @param hosts the hosts.
+     * @param allow when true allow hosts from the provided list to be connected to 
+     * (whether they will get connected to or not depends on the child policy), otherwise forbids these hosts
+     */
+    public static HostFilterPolicy mkHostAddressListPolicy( LoadBalancingPolicy childPolicy, Collection<InetSocketAddress> hosts, boolean allow) {
+        Predicate<Host> predicate = mkHostAddressPredicate(hosts);
+        return new HostFilterPolicy(childPolicy, allow ? predicate : Predicates.not(predicate));
+    }
+    
+    /**
+     * Create a new policy that wraps the provided child policy but only "allow/forbids" hosts
+     * whose DC belongs to the provided list.
+     *
+     * @param childPolicy the wrapped policy.
+     * @param dcs the DCs.
+     * @param allow when true allow hosts whose DC from the provided list to be connected to 
+     * (whether they will get connected to or not depends on the child policy), otherwise forbids these hosts
+     */
+    public static HostFilterPolicy mkDCListPolicy( LoadBalancingPolicy childPolicy, Collection<String> dcs, boolean allow) {
+        Predicate<Host> predicate = mkHostDCPredicate(dcs, allow);
```

Comment:
Yes. 
In case the DC of an address is unknown you want the behaviour of the policy to be the same independantly of the black or white list context.
Filtering is used in init and at this stage addresses may not be related to a DC.
(for example when using a VIP to access C* nodes)
You don't want these addresses to be filtered out thus the true value.